### DISCUSSION TITLE
questions concerning change notices

### DISCUSSION THREAD

**Discussion by gabrieledall** [OP-TED/eForms-SDK#88] [Q&A] [answered]
Trying to handle change notices, I have two questions to which I could't find an answer in the current documentation:

1. Notice Identifiers and changed Notice-Version reference ids
Change notices refer to the "changed notices" by referencing their Notice-Id (BT-701) and version ID  (BT-757) in one single BT-758 (Changed Notice Version Identifier). Since the change notice will contain the complete new notice (with all changes) as a whole, it remains unclear so far, how the notice-Identifiers of the complete changed document are expected to look like: will they remain similar to the Notice-Id of the  former Notice and will only the version-id be set up or is it expected to send a complete new version Id for the changed notice? The currently given examples present differnt notice Ids in BT 758 and in BT-701. Since we would perfer staying with the same notice-id in every change and only changing the version-id, we wonder, if there could be certain circumcances that would not allow this. Otherwise how would we / you able to manage more than one changes of a once changed notice - when each notice will get a new id and starts with version 1, especially when transmission errors lead to the situation that one change had not reached TED before another further change has arrived?

2. BT-13716: Change Previous Notice Section Identifier - will there be any standardized sections Identifiers and where will they be documented? 
Since some examples use e.g. "PROCEDURE" as reference to the "procedure"-section and LOT-0001 to refer to the first lot-section, I wonder if these section-identifiers are fixed and will be documented where this documentation will be accessible?

**Comment by YvesJo** [comment, marked as answer]
Hi Gabriele,
Please find the answers below:

**1. Notice Identifiers and changed Notice-Version reference ids**
A Change notice, is a notice on its own and requires its own Notice Identifier. The versioning is defined to allow for multiple successive editions until it gets published. Controls will be put in place to ensure, in the case of conflicts, that the latest version (i.e. the one with the highest version number) is considered. The association of a Change notice to its parent notice is performed using BT-758. There may be multiple changes applied in a single change notice (each change refers to the relevant section using BT-13716). When changes appear at different points in time, then successive Change notices have to be submitted, each referring to the previous one.

Changes may only be applied on published notices, therefore, the use case where a second change should be applied while the first one has not been published should be addressed either way:

- Complete and submit the first Change notice to have it published and then proceed with the second,
- integrate all changes in a single valid Change notice

When the non publication of the first Change is purely associated to non reliable transmission, then, if the first Change has to be published separately, use an alternate channel (e.g. eNotices2).

**2. BT-13716: Change Previous Notice Section Identifier** 

The BT-13716: Change Previous Notice Section Identifier refers to sections of the published notice. These reference identifiers should match identifiers that exist in the change notice. The list of section identifiers is reported in table 3 of [Referring to sections of a notice](https://docs.ted.europa.eu/eforms/1.0.0/schema/identifiers.html#_referring_to_sections_of_a_notice). 

LOT-XXXX refers in fact to the technical identifier for the Lot in which a change is made, for example LOT-0012. 

XXXX should not be understood as an order of occurrence and just seen as a univocal way to identify an object.

**Reply by alexmohora** [reply]
Hello,

So, for a contract notice an eSender may submit the contract Notice as A.01 (A being the ID and 01 the version) and then:
B.01, B.02, B.03 some change notices, B.01 changing and referring A.01, B.02 changing and referring B.01, B03 changing B.02. 
Is this correct?
All these change notices should have the same ID (B) and incremental versions? B <> A.

And if we have a contract award notice it is possible to have A.01 - with one contract (contract 1 for lot 1), A.02 when publishing a new contract with two contracts: the initial for lot 1 (contract 1) + another for lot 2 (contract 2).

If we have some change notices for the contract award notice it may be a situation like B.01, B.02,  B.03. 

B.01 change in contract 1 before publishing A.02: B.01 refers A.01
B.02 change in contract 2: B.02 refers A.02 not B.01(because B.01 does not contain contract 2)
but if B.02 contains another change in contract 1 after publishing A.02 in this case B.02 refers B.01 or A.02? A.02 includes the change B.01 in the contract 1 because is published after B.01.

Can you please comment on these situations or give some examples!

Thanks a lot
Alex Mohora

**Reply by alexmohora** [reply]
Just to add my opinion: 

I think the change notice should be considered a version of the same notice not a different notice because it's not a different notice. It contains exactly the same business information as the original notice but updated (or corrected) has the same structure + a specific section that describes changes. In this view all the notice versions in a set should have the same ID and incremental version. An instance should be unique identified by ID+VersionID. 

In this way we'll know that it is the same notice just different version - different version because data was updated either because of changes or because new data is to be published (new contracts in the contract award notice for example).

**Reply by YvesJo** [reply]
Hi Alex,
Thanks for sharing your questions.

**Changes & notice IDs**

> "All these change notices should have the same ID (B) and incremental versions? B <> A."

Each notice, should it be a change, shall have its own notice ID, so when successive changes are applied, they will not share the same "B". The version number is important until publication and when for a given notice ID a version gets published, no further version will be accepted.
So when there is a notice A subject to a change B, itself subject to a change C, we will have A => B => C ... each change notice making reference to the notice it changes (i.e. the one just before).

**Multiple CANs within a given procedure with contracts**
When I properly understand the scenario, we have an initial notice A, from there we have:

- a first CAN B with a contract on Lot 1
- a second CAN C with a contract on Lot 1 and another on lot 2

I see that Notice B and C both have contracts on a same lot (lot 1), this is a very specific situation like contracts awarded within a framework agreement. Also in this case, although the Contracts in both notices share a same lot, they aren't the same contracts and all contracts should be identified separately (Contract 1 on lot1 for notice B, contract 2 on lot1 for notice C and Contract 3 on lot 2 for notice C - given that the last ones are not grouped together in a single contract)

**Changes and CANs**
Changes apply on a whole notice, not on a contract and the change contains the whole notice consolidated (plus some additional fields specific to the change like reason, description, changed sections ...). Changes shall be seen as equivalent to corrigenda, they aren't there for contract modifications which are addressed using forms 38 to 40. So for successive changes we keep the sequence A => B => C.
Situations will be different with contract modifications as these notices will only contain the modified contracts.

KR,
Yves

**Reply by YvesJo** [reply]
Hi Alex,

> I think the change notice should be considered a version of the same notice not a different notice because it's not a different notice

I had the same view at start. It is indeed a similar however different notice. Intent is the same, a large part of the content will be the same ... however it is published at a different date and part of the content will differ. Versions are mainly useful during edition and to make sure the latest version is processed, especially as the notice reception is highly dependent on the transmission and processing channels involved

**Reply by alexmohora** [reply]
Thank you so much for your clarifications, for the change notice regarding contract notice it's clear for me.

But there is another issue with CAN. Maybe it's a business one... 

Let's say we have a procurement procedure with Lot 1 and Lot 2 and it's not a framework agreement. We have a just one contract notice CN (it can be only one per procedure). The procedure is finalized.

Then the contracting authority signs contract no 1 related with lot 1. It will publish in the national system a CAN with this contract (no1). Our system(eSender) send to TED a CAN A.01 (ID: A version 01). This CAN A.01 is successfully published in TED.

Two months later the contracting authority signs another contract, no2 related with lot2 for the same procedure. In our national system the CA access the procedure, then the CAN section and adds a new contract.

Our system prepares a new XML for TED
Is should be:
1. A new version of the initial CAN (A.02)
OR
3. A new CAN let's say B.01 for the same procedure (same procedureID as the CAN A.01)

IF 1 then CAN A.02 should contain both contracts (All signed contracts until that moment) - is consolidating the information or only the new one (contract no2)
IF 2 Then CAN B.01 should contain both contracts or only the new one.

What I'm trying to say is if the last CAN notice sent to TED should contain all the contracts for that procedure or only the new ones since the last TED publication? 

In the current system we send to TED all contracts because we have a section II.1.7) Total value of the procurement  that is computed from contracts values from section V.

**Reply by YvesJo** [reply]
In the latest CAN it should only be made reference to awarded contracts for which contract information were not previously published and the amounts should only refer to these contracts. 
Also when a procedure is ongoing and decision has not been made for some lots, these should also be identified at LotResult with the Winner selection status set to "open-nw"; of course there will be no contract for these.

**Reply by gabrieledall** [reply]
> Let's say we have a procurement procedure with Lot 1 and Lot 2 and it's not a framework agreement. We have a just one contract notice CN (**it can be only one per procedure**). The procedure is finalized.

This is an interesting topic I found here: the expectation, that only one (valid) CN is allowed to exist in a procedure. To me, this restriction comes from the process-oriented perspective regarding (only one) procurement process. Although this would be the usual case, there might be some exceptions to the rule:

Considering a long lasting project that will in itself be cut into (time-related) segments - each of them will lead to a single procurement process: At the moment each procurement will be regarded on it's own.

To me, it seems that the data driven point of view in eForms would also allow the publication of serveral CNs (dealing with different needs at different times if all of these belong to the same - long lasting - project):  it is not the notices that are relevant but the correct identification of all lots, contracts, etc. within the procedure that define the outcomes of the whole project. 

Does my understanding fit to the underlying data concept in eForms or is this a kind of over-interpretation?

Kind Regards
Gabriele

**Reply by YvesJo** [reply]
> Considering a long lasting project that will in itself be cut into (time-related) segments - each of them will lead to a single procurement process: At the moment each procurement will be regarded on it's own.

Yes and each of these procurement will be seen as a different procedure,

> To me, it seems that the data driven point of view in eForms would also allow the publication of serveral CNs (dealing with different needs at different times if all of these belong to the same - long lasting - project): it is not the notices that are relevant but the correct identification of all lots, contracts, etc. within the procedure that define the outcomes of the whole project.

I agree with the statement and if I understand it properly it would start from a planning notice and lead to multiple competition notices. This is perfectly exact, as illustration we may also remind that not only Parts from PIN only notices may be combined to define a competition notice but a given Part may lead to multiple Competition notices. We shall also keep in mind that a Procedure actually starts with a competition notice (NB: there is no procedure ID associated to planning notices), planning notices are not _stricto sensu_ part of the Procedure. So I think we agree and the overall process covers everything from Planning to Result with many interrelated notices possibilities and eventually covers multiple procedures, each starting with their respective Competition notices.

**Reply by EmilyBrasa** [reply]
I have some followup questions on this topic. 

In the FAQ it says: “Each time a notice with the same notice identifier is submitted and accepted by the destination application (and publication is not stopped by the user), it must have a different version ID (starting at "01" and incrementing). “

1.	If for whatever reason a version is missing in the chain (e.g. for A we post versionID A.01, A.02 and A.04) would TED accept the latest version A.04 or reject it because version A.03 is missing?

2.	Would it even be possible to post the same noticeID we post A.01, A.04 and afterwards A.02? If yes, which version would then be published – the highest (A.04) or the most recently submitted (A.02)?

3.	What about stopping then? If for the same noticeID we post A.01, A.02 and A.03, but the Stop A.03 – Do we have to send A.04 afterwards or would it be possible to send A.03 again replacing the stopped A.03?

Thanks in advance

**Reply by YvesJo** [reply]
Hi,
Your question has previously reached the Helpdesk and is under process. As a general rule, non-technical questions should be directly sent to the Helpdesk: ted@publications.europa.eu
Also for technical questions, preferably start a separate discussion instead of adding comments to answered ones as it would prevent it to not be detected.
Thanks for your understanding.
